### PR TITLE
Update tuples topic for C# 7.1

### LINKED ISSUE #2798: Update tuples topics for Inferred tuple element names
C# 7.1 enhances the tuples feature to [infer names of tuple elements](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.1/infer-tuple-names.md) which makes them easier to declare.

To handle this, we need to:

- [x] Update the Tuples topic to show this syntax
- [x] review deconstruction to make sure it is still accurate.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #2798

This update discusses tuple field name projections.

### CODE REVIEW

**Comment by BillWagner** [comment]
Would you like to look over this one as well @jcouv?

File: docs/csharp/tuples.md [hunk 1]
```diff
@@ -85,6 +81,9 @@ unnamed tuple:
 
 [!code-csharp[UnnamedTuple](../../samples/snippets/csharp/tuples/tuples/program.cs#01_UnNamedTuple "Unnamed tuple")]
 
+The tuple in the above example was initialized using literal constants and
```
Comment (mairaw) [inline docs/csharp/tuples.md]: above -> previous

File: docs/csharp/tuples.md [hunk 2]
```diff
@@ -85,6 +81,9 @@ unnamed tuple:
...
+will not have field names created using *Tuple field name projections* in C# 7.1.
```
Comment (mairaw) [inline docs/csharp/tuples.md]: will not -> won't

File: docs/csharp/tuples.md [hunk 3]
```diff
@@ -117,6 +122,39 @@ It is important to understand these underlying fundamentals of
 the new tuples and the `ValueTuple` type in order to understand
 the rules for assigning named tuples to each other.
 
+## Tuple projection initializers
+
+In general, tuple projection initializers work by using the variable or
```
Comment (mairaw) [inline docs/csharp/tuples.md]: right-hand should be hyphened

**Review by mairaw — COMMENTED** [review]
minor edits

**Comment by BillWagner** [comment]
@mairaw I fixed everything you mentioned. Ready to :shipit:?

File: docs/csharp/tuples.md [hunk 4]
```diff
@@ -42,30 +42,26 @@ different ways to use them, and initial guidance on working with Tuples.
 
 Let's start with the reasons for adding new Tuple support. Methods return
 a single object. Tuples enable you to package multiple values in that single
-object more easily. 
+object more easily.
 
 The .NET Framework already has generic `Tuple` classes. These classes,
 however, had two major limitations. For one, the `Tuple` classes named
 their fields `Item1`, `Item2`, and so on. Those names carry no semantic
```
Comment (jcouv) [inline docs/csharp/tuples.md]: Those where properties, not fields.

File: docs/csharp/tuples.md [hunk 5]
```diff
@@ -42,30 +42,26 @@ different ways to use them, and initial guidance on working with Tuples.
...
 information. Using these `Tuple` types does not enable communicating the
-meaning of each of the fields. Another concern is that the `Tuple` classes are
+meaning of each of the fields. The new language features enable you to declare
+and use semantically meaningful names for the fields in a tuple.
```
Comment (jcouv) [inline docs/csharp/tuples.md]: Perhaps "elements" instead of "fields".

File: docs/csharp/tuples.md [hunk 6]
```diff
@@ -42,30 +42,26 @@ different ways to use them, and initial guidance on working with Tuples.
...
+
+Another concern is that the `Tuple` classes are
 reference types. Using one of the `Tuple` types means allocating objects. On hot
-paths, this can have a measurable impact on your application's performance.
+paths, this can have a measurable impact on your application's performance. Therefore,
+the language support for tuples leverages the new `ValueTuple` classes.
```
Comment (jcouv) [inline docs/csharp/tuples.md]: Perhaps "types" or "structs" instead of "classes".

File: docs/csharp/tuples.md [hunk 7]
```diff
@@ -85,6 +81,9 @@ unnamed tuple:
 
 [!code-csharp[UnnamedTuple](../../samples/snippets/csharp/tuples/tuples/program.cs#01_UnNamedTuple "Unnamed tuple")]
 
+The tuple in the previous example was initialized using literal constants and
+won't have field names created using *Tuple field name projections* in C# 7.1.
```
Comment (jcouv) [inline docs/csharp/tuples.md]: >field [](start = 11, length = 5)

General terminology: we refer to "elements" of tuples, and the "cardinality" of tuples (for the number of elements).
ValueTuple on the other hand has fields and an arity (from 0 to 8).

File: docs/csharp/tuples.md [hunk 8]
```diff
@@ -267,6 +305,10 @@ expresses that design very well:
 
 [!code-csharp[QueryReturningTuple](../../samples/snippets/csharp/tuples/tuples/projectionsample.cs#15_QueryReturningTuple "Query returning a tuple")]
 
+> [!NOTE]
+> In C# 7.1, tuple projections enable you to create named tuples using fields, in a manner simlar to the property naming in anonymous types. In the above code,
```
Comment (jcouv) [inline docs/csharp/tuples.md]: >simlar [](start = 91, length = 6)

s/simlar/similar/

**Review by jcouv — APPROVED** [review]
LGTM with a couple of nits and suggestions.

File: docs/csharp/tuples.md [hunk 9]
```diff
@@ -82,7 +82,7 @@ unnamed tuple:
 [!code-csharp[UnnamedTuple](../../samples/snippets/csharp/tuples/tuples/program.cs#01_UnNamedTuple "Unnamed tuple")]
 
 The tuple in the previous example was initialized using literal constants and
-won't have field names created using *Tuple field name projections* in C# 7.1.
+won't have element names created using *Tuple field name projections* in C# 7.1.
```
Comment (jcouv) [inline docs/csharp/tuples.md]: You may want to do a scrub on the whole file for "fields". For instance, here you can say "Tuple element name projections".

**Review by jcouv — APPROVED** [review]
Awesome. Thanks!

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks good, @BillWagner. Left a few comments.

**Comment by BillWagner** [comment]
@rpetrusha I addressed all your comments. Do you want to give it a final look?

**Comment by rpetrusha** [comment]
@BillWagner, LGTM. I'll merge.

**Review by rpetrusha — APPROVED** [review]

**Pull request merged**